pkg/runtime/executor.go

```diff
@@ -290,3 +306,12 @@ func implementsSubFolderScan(iacType string) bool {
 	}
 	return false
 }
+
+// makeNonRecursiveTrue sets the 'NonRecursiveScan' field to true
+// only terraform iac provider should be passed to this function, else it will panic
+func makeNonRecursiveTrue(iacP iacProvider.IacProvider) iacProvider.IacProvider {
```

Comment:
Do you think, it would be better to have non-recursive as part of the IacProvider interface rather than setting an option using reflection package.